src/som/interpreter/actors/ReceivedMessage.java

```diff
@@ -35,9 +36,12 @@ public Object execute(final VirtualFrame frame) {
       dbg.prepareSteppingAfterNextRootNode();
     }
 
-    Object result = onReceive.doPreEvaluated(frame, msg.args);
-
-    resolvePromise(frame, msg.resolver, result, msg.triggerPromiseResolutionBreakpoint);
+    try {
+      Object result = onReceive.doPreEvaluated(frame, msg.args);
+      resolvePromise(frame, msg.resolver, result, msg.triggerPromiseResolutionBreakpoint);
+    } catch (SomException exception) {
+      msg.getResolver().onError(exception);
```

Comment:
I think I would prefer that you do `exception.getSomObject()` here. So, that `onError` does not need to deal with this special case.